src/lib/api/documents.test.ts

```diff
@@ -12,14 +21,88 @@ const test = base.extend({
       "./fixtures/documents/document-expanded.json"
     );
 
-    await use(document as unknown as Document);
+    await use(document as Document);
+  },
+
+  created: async ({}, use: Use<Document[]>) => {
+    const { default: created } = await import(
+      "./fixtures/documents/create.json"
+    );
+
+    await use(created as Document[]);
   },
 });
 
 describe("document fetching", () => {
-  test.todo("get a single document");
+  test.todo("documents.get");
+  test.todo("documents.search");
+});
+
+describe("document uploads and processing", () => {
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  test("documents.create", async ({ created }) => {
+    const mockFetch = vi.fn().mockImplementation(async () => ({
+      ok: true,
+      async json() {
+        return created;
+      },
+    }));
+
+    const docs: DocumentUpload[] = created.map((d) => ({
+      title: d.title,
+      access: d.access,
+      language: d.language,
+      original_extension: d.original_extension,
+    }));
+
+    documents.create(docs, "token", mockFetch).then((d) => {
+      expect(d).toMatchObject(created);
+    });
+  });
+
+  test("documents.upload", async ({ created }) => {
+    const mockFetch = vi.fn().mockImplementation(async () => ({
+      ok: true,
+    }));
+
+    created.forEach(async (doc) => {
+      const file = new File(
+        ["test content"],
+        "finalseasonal-allergies-pollen-and-mold-2023-en.pdf",
+        { type: "application/pdf" },
+      );
+
+      const resp = await documents.upload(
+        new URL(doc.presigned_url),
+        file,
+        mockFetch,
+      );
+
+      expect(resp.ok).toBeTruthy();
```

Comment:
In these tests, I'd also assert that mockFetch is called with the expected properties and values. This way, we can ensure that these functions make valid API calls.

```suggestion
      expect(resp.ok).toBeTruthy();
      const expectedEndpoint = ...;
      const expectedOptions = ...;
      expect(mockFetch).toHaveBeenCalledWith(expectedEndpoint, expectedOptions);
```